PsfLauncher/main.cpp

```diff
@@ -268,43 +277,233 @@ void LaunchMonitorInBackground(std::filesystem::path packageRoot, const wchar_t
+            std::wostringstream ss;
+            ss << L"Could not initilize the proc thread attribute list.";
+
+            ErrorInformation error(ss.str(), err);
+            return error;
+        }
+
+        DWORD attribute = PROCESS_CREATION_DESKTOP_APP_BREAKAWAY_DISABLE_PROCESS_TREE;
+        if (UpdateProcThreadAttribute(StartupInfoEx.lpAttributeList,
+            0,
+            PROC_THREAD_ATTRIBUTE_DESKTOP_APP_POLICY,
+            &attribute,
+            sizeof(attribute),
+            NULL,
+            NULL) == FALSE)
+        {
+            auto err{ ::GetLastError() };
+            std::wostringstream ss;
+            ss << L"Could not update Proc thread attribute.";
+
+            ErrorInformation error(ss.str(), err);
+            return error;
+        }
+    }
+
+    PROCESS_INFORMATION processInfo = { 0 };
+    if (::CreateProcessW(
+        execInfo.ApplicationName,
+        execInfo.CommandLine.data(),
+        nullptr, nullptr, // Process/ThreadAttributes
+        true, // InheritHandles
+        EXTENDED_STARTUPINFO_PRESENT, // CreationFlags
+        nullptr, // Environment
+        execInfo.CurrentDirectory,
+        (LPSTARTUPINFO)&StartupInfoEx,
+        &processInfo))
+    {
+        // Propagate exit code to caller, in case they care
+        DWORD waitResult = ::WaitForSingleObject(processInfo.hProcess, INFINITE);
+
+        if (waitResult == WAIT_OBJECT_0)
+        {
+            ErrorInformation noError;
+            return noError;
         }
         else
         {
-            if (wait)
-                WaitForSingleObject(processInfo.hProcess, INFINITE);
+            auto err{ ::GetLastError() };
+            std::wostringstream ss;
+            ss << L"Running process failed.";
+
+            ErrorInformation error(ss.str(), err);
+            return error;
+        }
+    }
+    else
+    {
+        std::wostringstream ss;
+        auto err{ ::GetLastError() };
+
+        if (execInfo.ApplicationName)
+        {
+            ss << L"ERROR: Failed to create a process for " << execInfo.ApplicationName << " ";
+        }
+        else
+        {
+            std::wstring applicationNameForError;
+            applicationNameForError = execInfo.CommandLine.substr(0, execInfo.CommandLine.find(' '));
+            ss << L"ERROR: Failed to create a process for " << applicationNameForError << " ";
         }
+
+        ErrorInformation error(ss.str(), err);
+        return error;
     }
 }
-int __stdcall wWinMain(HINSTANCE, HINSTANCE, PWSTR args, int cmdShow)
+
+ErrorInformation StartWithShellExecute(std::filesystem::path packageRoot, std::filesystem::path exeName, std::wstring exeArgString, LPCWSTR dirStr, int cmdShow)
 {
-    return launcher_main(args, cmdShow);
+    // Non Exe case, use shell launching to pick up local FTA
+    auto nonexePath = packageRoot / exeName;
+
+    CoInitializeEx(NULL, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE);
+    SHELLEXECUTEINFO shex;

```

Comment:
why is this needed?

Reply:
Because the monitor program will run on a separate thread.  This is used only for PSFFixupMonitor.  But, now that I'm thinking about it, we only need it for the monitor.  I'm going to move this up to main and only invoke CoInitilizeEx if we have a monitor we need to run.